### PR TITLE
Force fs to use rebar3 manager to fix error. See sync/fs#33

### LINKED ISSUE #65: Error when code reloader enabled
### Environment

* Elixir / Hex version (mix hex.info):

```
$ mix hex.info
Hex:    0.14.1
Elixir: 1.3.3
OTP:    19.1

Built with: Elixir 1.3.4 and OTP 18.3.4.4
```

* Node / NPM version (node -v) (npm -v):

```
$ node -v
v6.9.1
$ npm -v
3.10.8
```

* Operating system:

```
$ uname -a
Darwin Compe.local 16.1.0 Darwin Kernel Version 16.1.0: Thu Oct 13 21:26:57 PDT 2016; root:xnu-3789.21.3~60/RELEASE_X86_64 x86_64
```

### Current behavior

When running a brand new Kitto dashboard installed from the 0.3.0 archive, opening the sample dashboard raises errors in the console:

```
16:10:40.726 [error] Ranch listener Kitto.Router.HTTP had connection process started with :cowboy_protocol:start_link/4 at #PID<0.1402.0> exit with reason: {{:noproc, {GenServer, :call, [nil, {:get, #Function<1.103310966/1 in Kitto.Notifier.cache/0>}, 5000]}}, {Kitto.Router, :call, [%Plug.Conn{adapter: {Plug.Adapters.Cowboy.Conn, :...}, assigns: %{}, before_send: [], body_params: %Plug.Conn.Unfetched{aspect: :body_params}, cookies: %Plug.Conn.Unfetched{aspect: :cookies}, halted: false, host: "localhost", method: "GET", owner: #PID<0.1402.0>, params: %Plug.Conn.Unfetched{aspect: :params}, path_info: ["events"], path_params: %{}, peer: {{127, 0, 0, 1}, 51690}, port: 4000, private: %{}, query_params: %Plug.Conn.Unfetched{aspect: :query_params}, query_string: "topics=text,random,buzzwords,random,convergence", remote_ip: {127, 0, 0, 1}, req_cookies: %Plug.Conn.Unfetched{aspect: :cookies}, req_headers: [{"host", "localhost:4000"}, {"connection", "keep-alive"}, {"accept", "text/event-stream"}, {"cache-control", "no-cache"}, {"user-agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.75 Safari/537.36"}, {"referer", "http://localhost:4000/dashboards/sample"}, {"accept-encoding", "gzip, deflate, sdch, br"}, {"accept-language", "en-US,en;q=0.8"}], request_path: "/events", resp_body: nil, resp_cookies: %{}, resp_headers: [{"cache-control", "max-age=0, private, must-revalidate"}], scheme: :http, script_name: [], secret_key_base: nil, state: :unset, status: nil}, []]}}
```

No data is rendered to the dashboard.

Workaround is to disable code reloading by appending the following code to `config/config.exs`:

```elixir
config :kitto, reload_code?: false
```

I've been able to replicate the error from the master branch as well.

### Expected behavior

I expect to not have errors in the console from a base install.

### PR DESCRIPTION
**Pull request by davejlong**
Fixes #65. An unreleased version of `fs` forces the `rebar3` manager to compile the dependency. See https://github.com/synrc/fs/issues/33

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -32,7 +32,7 @@ defmodule Kitto.Mixfile do
     [{:cowboy, "~> 1.0.0"},
      {:plug, "~> 1.3"},
      {:poison, "~> 3.0"},
-     {:fs, "~> 2.11.0"},
+     {:fs, github: "synrc/fs"},
```
Comment (zorbash) [inline mix.exs]: You can't release a hex package with a non-packaged dependency :-/

Comment (zorbash) [reply mix.exs]: Can you try this? https://github.com/synrc/fs/issues/33#issuecomment-265483357

Comment (davejlong) [reply mix.exs]: I did. It doesn't work. Needs the new `fs` updates on the `master` branch to be released.

Comment (zorbash) [reply mix.exs]: Honestly with the repo not having git tags to match the released versions its quite hard to figure out which are the unreleased updates you're referring to. I prefer to downgrade to 0.9.2 than re-add a github dependency.

I've commented on the lack of git tags, see https://github.com/synrc/fs/issues/24#issuecomment-266221803.

Comment (davejlong) [inline mix.exs]: What do you think about maybe dropping in https://github.com/falood/exfswatch as a replacement for fs? Seems to have more documentation and tagging.

Comment (zorbash) [reply mix.exs]: @davejlong did you delete a comment here?
I have this on my feed, strange..
selection_076

Comment (davejlong) [reply mix.exs]: I did. I looked further into exfswatch and it just seems to be a layer on top of fs so we'd end up with the same issues.

Comment (zorbash) [reply mix.exs]: @davejlong right i considered it in the beginning, it'd be the same.

**Comment by zorbash** [comment]
The best thing to do now is to raise some awareness on https://github.com/synrc/fs/issues/24 and wait. If there's no response we should consider pushing a package from our fork.

**Comment by davejlong** [comment]
I started playing with writing a fairly simple wrapper for the `fswatch` library. If it works, setup would be easier as `fswatch` is a pure C implementation which works on Mac, *nix, Microsoft.

**Comment by zorbash** [comment]
@davejlong Can you confirm that fs@2.12 works on the mac?

**Comment by zorbash** [comment]
We have a confirmation that 2.12.0 fixes the issue. See: https://github.com/kittoframework/kitto/issues/65#issuecomment-268254803

Merging this. Thanks @davejlong for the fantastic work on finding the root cause of this issue and doing whatever needed to resolve it. :clap: :clap:

**Comment by zorbash** [comment]
@davejlong I've seen [fswatch](https://github.com/emcrisostomo/fswatch#) and it seems like a really promising attempt to make fs notifications cross-platform. What bothers me it's lack of packaging for Linux (see: https://github.com/emcrisostomo/fswatch#installation). No deb packages yet :-/

Also there's a change that fswatch will be supported through fs, see: https://github.com/synrc/fs/issues/17

**Pull request merged**